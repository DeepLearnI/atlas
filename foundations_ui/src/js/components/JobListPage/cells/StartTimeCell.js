import React, { Component } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import HoverCell from './HoverCell';
import JobListActions from '../../../actions/JobListActions';
import CommonActions from '../../../actions/CommonActions';


class StartTimeCell extends Component {
  constructor(props) {
    super(props);
    this.toggleExpand = this.toggleExpand.bind(this);
    this.state = {
      date: JobListActions.getFormatedDate(this.props.startTime),
      time: JobListActions.getFormatedTime(this.props.startTime),
      isError: this.props.isError,
      rowNumber: this.props.rowNumber,
      expand: this.props.expand,
      status: this.props.status,
    };
  }

  toggleExpand(value) {
    this.setState({ expand: value });
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.rowNumber !== this.props.rowNumber) {
      this.setState({
        time: JobListActions.getFormatedTime(nextProps.startTime),
        date: JobListActions.getFormatedDate(nextProps.startTime),
        rowNumber: nextProps.rowNumber,
        isError: nextProps.isError,
        expand: nextProps.expand,
        status: nextProps.status,
      });
    }
  }


  render() {
    const {
      date, time, isError, rowNumber, expand, status,
    } = this.state;
    let hover;

    const errorClass = CommonActions.errorStatus(isError);
    const spanClass = ''.concat(errorClass);

    const pClass = isError
      ? `job-cell start-cell error row-${rowNumber}`
      : `job-cell start-cell row-${rowNumber}`;

    const launchDate = status === 'queued' ? 'Queued' : moment(date).format('MMM DD').toString();
    const launchTime = status === 'queued' ? '' : time;

    const dateTimeFormatted = (
      <span className="font-bold">
        <span className="launch-date">{launchDate} </span>
        <span className={spanClass}>{launchTime}</span>
      </span>
    );

    if (expand && date !== '') {
      hover = <HoverCell textToRender={dateTimeFormatted} />;
    }

    return (
      <span
        key={`${date}-${time}`}
        onMouseEnter={() => this.toggleExpand(true)}
        onMouseLeave={() => this.toggleExpand(false)}
      >
        {dateTimeFormatted}
        <span>{hover}</span>
      </span>
    );
  }
}

StartTimeCell.propTypes = {
  startTime: PropTypes.string,
  isError: PropTypes.bool,
  rowNumber: PropTypes.number,
  expand: PropTypes.bool,
  status: PropTypes.string,
};

StartTimeCell.defaultProps = {
  startTime: '',
  isError: false,
  rowNumber: -1,
  expand: false,
  status: '',
};

export default StartTimeCell;
