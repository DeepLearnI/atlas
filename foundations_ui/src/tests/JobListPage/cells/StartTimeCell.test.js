import React from 'react';
import ReactDOM from 'react-dom';
import StartTimeCell from '../../../js/components/JobListPage/cells/StartTimeCell';
import { shallow } from 'enzyme';
import configureTests from '../../setupTests';

configureTests();

it('Shallow Renders Start Time Cell', () => {
  const wrapper = shallow(<StartTimeCell/>);
  expect(wrapper).toMatchSnapshot();
});

it('Sets Expand State Upon Hover', () => {
  const wrapper = shallow(<StartTimeCell/>);
  wrapper.find('div').first().simulate('mouseEnter');
  expect(wrapper.state("expand")).toEqual(true);
  wrapper.find('div').first().simulate('mouseLeave');
  expect(wrapper.state("expand")).toEqual(false);
});