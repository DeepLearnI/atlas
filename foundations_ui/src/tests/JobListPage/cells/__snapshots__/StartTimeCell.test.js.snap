// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Renders Start Time Cell 1`] = `
<p
  className="job-cell header-4 font-bold "
>
   
  <span
    className="font-regular "
  />
</p>
`;
